helper/tests/testing.go

```diff
@@ -120,6 +125,48 @@ func WaitUntilTxPoolEmpty(ctx context.Context, client txpoolOp.TxnPoolOperatorCl
 	return status, nil
 }
 
+func WaitForNonce(
+	ctx context.Context,
+	ethClient *jsonrpc.Eth,
+	addr web3.Address,
+	expectedNonce uint64,
+) (
+	interface{},
+	error,
+) {
+	type result struct {
+		nonce uint64
+		err   error
+	}
+
+	resObj, err := RetryUntilTimeout(ctx, func() (interface{}, bool) {
+		nonce, err := ethClient.GetNonce(addr, web3.Latest)
+		if err != nil {
+			//	error -> stop retrying
+			return result{nonce, err}, false
+		}
+
+		if nonce == expectedNonce {
```

Comment:
What do you think about flipping this conditional to `>=` instead of `==`?

That way, even if you grab a higher nonce, it's considered valid